### PR TITLE
Digest #21

### PR DESCRIPTION
**Pull request by pjazdzewski1990**
2015-03-04 16:10:00

### CODE REVIEW

File: _posts/04-03-2015-digest21.md [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+
+
+<p id="1"></p>
+* \[Scala\] [Exploring Scala Macros: map to case class conversion](http://blog.echo.sh/2013/11/04/exploring-scala-macros-map-to-case-class-conversion.html) 
+* \[Scala\] [Akka's Reactive Stream Implementation](http://bryangilbert.com/blog/2015/02/04/akka-reactive-streams/index.html)
+* \[Architecture\] [Adopting Microservices at Netflix: Lessons for Architectural Design](http://nginx.com/blog/microservices-at-netflix-architectural-best-practices/)
+* \[Functional Programming\] [Monoids, Functors, Applicatives, and Monads: 10 Main Ideas](https://monadmadness.wordpress.com/2015/01/02/monoids-functors-applicatives-and-monads-10-main-ideas/)
+* \[Frontend\] [Frontend Guidelines](https://github.com/bendc/frontend-guidelines)
+* \[Mobile\] [How NativeScript Works](http://developer.telerik.com/featured/nativescript-works/)
```
Comment (marioosh) [inline _posts/04-03-2015-digest21.md]: :-1: not scala, not functinal, just tool. Commercial, non free tool.

File: _posts/04-03-2015-digest21.md [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+Hi folks!
+
+
+
+<p id="1"></p>
+* \[Scala\] [Exploring Scala Macros: map to case class conversion](http://blog.echo.sh/2013/11/04/exploring-scala-macros-map-to-case-class-conversion.html) 
+* \[Scala\] [Akka's Reactive Stream Implementation](http://bryangilbert.com/blog/2015/02/04/akka-reactive-streams/index.html)
+* \[Architecture\] [Adopting Microservices at Netflix: Lessons for Architectural Design](http://nginx.com/blog/microservices-at-netflix-architectural-best-practices/)
+* \[Functional Programming\] [Monoids, Functors, Applicatives, and Monads: 10 Main Ideas](https://monadmadness.wordpress.com/2015/01/02/monoids-functors-applicatives-and-monads-10-main-ideas/)
```
Comment (marioosh) [inline _posts/04-03-2015-digest21.md]: :+1:

File: _posts/04-03-2015-digest21.md [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+- Scalac Weekly Digest
+- Scala
+---
+
+Hi folks!
+
+
+
+<p id="1"></p>
+* \[Scala\] [Exploring Scala Macros: map to case class conversion](http://blog.echo.sh/2013/11/04/exploring-scala-macros-map-to-case-class-conversion.html)
```
Comment (marioosh) [inline _posts/04-03-2015-digest21.md]: :+1: fajne

File: _posts/04-03-2015-digest21.md [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+Hi folks!
+
+
+
+<p id="1"></p>
+* \[Scala\] [Exploring Scala Macros: map to case class conversion](http://blog.echo.sh/2013/11/04/exploring-scala-macros-map-to-case-class-conversion.html) 
+* \[Scala\] [Akka's Reactive Stream Implementation](http://bryangilbert.com/blog/2015/02/04/akka-reactive-streams/index.html)
+* \[Architecture\] [Adopting Microservices at Netflix: Lessons for Architectural Design](http://nginx.com/blog/microservices-at-netflix-architectural-best-practices/)
+* \[Functional Programming\] [Monoids, Functors, Applicatives, and Monads: 10 Main Ideas](https://monadmadness.wordpress.com/2015/01/02/monoids-functors-applicatives-and-monads-10-main-ideas/)
+* \[Frontend\] [Frontend Guidelines](https://github.com/bendc/frontend-guidelines)
```
Comment (szabba) [inline _posts/04-03-2015-digest21.md]: People bashed this one pretty hard on reddit.

http://www.reddit.com/r/programming/comments/2xkgw3/frontendguidelines_some_html_css_and_js_best/

File: _posts/04-03-2015-digest21.md (hunk 3 above)
Comment (pjazdzewski1990) [inline _posts/04-03-2015-digest21.md]: also a bit old ....

File: _posts/04-03-2015-digest21.md [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+* \[Mobile\] [How NativeScript Works](http://developer.telerik.com/featured/nativescript-works/)
+* \[Functional Programming\] [The Philosophy of Ramda](http://fr.umio.us/the-philosophy-of-ramda/)
+* \[Android\] [An Investigation into Flow and Mortar](http://www.bignerdranch.com/blog/an-investigation-into-flow-and-mortar/)
+* \[Startups\] [Startup advice, briefly](http://blog.samaltman.com/startup-advice-briefly)
+* \[Javascript\] [React Tips and Best Practices](http://aeflash.com/2015-02/react-tips-and-best-practices.html)
+* \[Functional Programming\] [Monads in Dynamically-Typed Languages](http://eighty-twenty.org/2015/01/25/monads-in-dynamically-typed-languages/)
```
Comment (pjazdzewski1990) [inline _posts/04-03-2015-digest21.md]: I think it's mostly a Racket thing and we don't use that language here ... it's not obvious to me how this knowledge could be passed in languages we use

File: _posts/04-03-2015-digest21.md (hunk 3 above)
Comment (note) [inline _posts/04-03-2015-digest21.md]: :+1: excellent

File: _posts/04-03-2015-digest21.md [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+- Scala
+---
+
+Hi folks!
+
+
+
+<p id="1"></p>
+* \[Scala\] [Exploring Scala Macros: map to case class conversion](http://blog.echo.sh/2013/11/04/exploring-scala-macros-map-to-case-class-conversion.html) 
+* \[Scala\] [Akka's Reactive Stream Implementation](http://bryangilbert.com/blog/2015/02/04/akka-reactive-streams/index.html)
```
Comment (note) [inline _posts/04-03-2015-digest21.md]: :+1: Very informative

File: _posts/04-03-2015-digest21.md (hunk 2 above)
Comment (note) [inline _posts/04-03-2015-digest21.md]: :+1:

File: _posts/04-03-2015-digest21.md [hunk 7]
```diff
@@ -0,0 +1,29 @@
...
+* \[Architecture\] [Adopting Microservices at Netflix: Lessons for Architectural Design](http://nginx.com/blog/microservices-at-netflix-architectural-best-practices/)
```
Comment (note) [inline _posts/04-03-2015-digest21.md]: The article itself seems laconic, havent't seen videos

File: _posts/04-03-2015-digest21.md [hunk 8]
```diff
@@ -0,0 +1,29 @@
...
+* \[Functional Programming\] [Monoids, Functors, Applicatives, and Monads: 10 Main Ideas](https://monadmadness.wordpress.com/2015/01/02/monoids-functors-applicatives-and-monads-10-main-ideas/)
+* \[Frontend\] [Frontend Guidelines](https://github.com/bendc/frontend-guidelines)
+* \[Mobile\] [How NativeScript Works](http://developer.telerik.com/featured/nativescript-works/)
+* \[Functional Programming\] [The Philosophy of Ramda](http://fr.umio.us/the-philosophy-of-ramda/)
+* \[Android\] [An Investigation into Flow and Mortar](http://www.bignerdranch.com/blog/an-investigation-into-flow-and-mortar/)
+* \[Startups\] [Startup advice, briefly](http://blog.samaltman.com/startup-advice-briefly)
```
Comment (note) [inline _posts/04-03-2015-digest21.md]: :-1: I am not a fan of posts like this - I don't believe that there's something like a recipe for a successfull startup.

File: _posts/04-03-2015-digest21.md (hunk 5 above)
Comment (szabba) [inline _posts/04-03-2015-digest21.md]: It's not a "take this and use it directly" kind of thing, but it should be easy to port to Clojure, and references other materials, so the reader can dig in deeper into the possible designs for monad libraries. But then, I'm not defending it, I just weren't sure if it'd be a good fit.

File: _posts/04-03-2015-digest21.md (hunk 3 above)
Comment (andrjew) [inline _posts/04-03-2015-digest21.md]: maybe old but very good if someone didnt play with macro :+1:

File: _posts/04-03-2015-digest21.md (hunk 2 above)
Comment (akaczynski) [inline _posts/04-03-2015-digest21.md]: a lot of algebra thingy, but it was nice

Comment (zakrzes) [inline _posts/04-03-2015-digest21.md]: :+1:

File: _posts/04-03-2015-digest21.md (hunk 1 above)
Comment (zakrzes) [inline _posts/04-03-2015-digest21.md]: might be interesting :+1:

File: _posts/04-03-2015-digest21.md [hunk 9]
```diff
@@ -0,0 +1,29 @@
...
+
+<p id="1"></p>
+* \[Scala\] [Exploring Scala Macros: map to case class conversion](http://blog.echo.sh/2013/11/04/exploring-scala-macros-map-to-case-class-conversion.html) 
+* \[Scala\] [Akka's Reactive Stream Implementation](http://bryangilbert.com/blog/2015/02/04/akka-reactive-streams/index.html)
+* \[Architecture\] [Adopting Microservices at Netflix: Lessons for Architectural Design](http://nginx.com/blog/microservices-at-netflix-architectural-best-practices/)
+* \[Functional Programming\] [Monoids, Functors, Applicatives, and Monads: 10 Main Ideas](https://monadmadness.wordpress.com/2015/01/02/monoids-functors-applicatives-and-monads-10-main-ideas/)
+* \[Frontend\] [Frontend Guidelines](https://github.com/bendc/frontend-guidelines)
+* \[Mobile\] [How NativeScript Works](http://developer.telerik.com/featured/nativescript-works/)
+* \[Functional Programming\] [The Philosophy of Ramda](http://fr.umio.us/the-philosophy-of-ramda/)
+* \[Android\] [An Investigation into Flow and Mortar](http://www.bignerdranch.com/blog/an-investigation-into-flow-and-mortar/)
```
Comment (zakrzes) [inline _posts/04-03-2015-digest21.md]: :+1:

File: _posts/04-03-2015-digest21.md [hunk 10]
```diff
@@ -0,0 +1,43 @@
...
+- Mobile
+- Android
+---
+
+Hi folks!
+
+<p id="1"></p>
+* \[Scala\] [Exploring Scala Macros: map to case class conversion](http://blog.echo.sh/2013/11/04/exploring-scala-macros-map-to-case-class-conversion.html) 
+
+The article above is a nice explanation of Macros. If you didn't use this useful feature of Scala before, here's a chance for you. And you can trust us, it's not not so hard as you might think.
```
Comment (marioosh) [inline _posts/04-03-2015-digest21.md]: **not not** so hard

Comment (mtomas) [inline _posts/04-03-2015-digest21.md]: contains nice explanation of macros.

File: _posts/04-03-2015-digest21.md [hunk 11]
```diff
@@ -0,0 +1,43 @@
...
+
+<p id="2"></p>
+* \[Scala\] [Akka's Reactive Stream Implementation](http://bryangilbert.com/blog/2015/02/04/akka-reactive-streams/index.html) 
+
+Reactive Streams are a hot topic right now. If you would like to start working with stream, this article is a great place to start. Bryan Gilbert explores the akka-stream library and in a few simple words explains who, how and why can benefit from this technology.
```
Comment (mtomas) [inline _posts/04-03-2015-digest21.md]: with streams

File: _posts/04-03-2015-digest21.md [hunk 12]
```diff
@@ -0,0 +1,43 @@
...
+
+<p id="2"></p>
+* \[Scala\] [Akka's Reactive Stream Implementation](http://bryangilbert.com/blog/2015/02/04/akka-reactive-streams/index.html) 
+
+Reactive Streams are a hot topic right now. If you would like to start working with stream, this article is a great place to start. Bryan Gilbert explores the akka-stream library and in a few simple words explains who, how and why can benefit from this technology. 
+
+<p id="3"></p>
+* \[Functional Programming\] [Monoids, Functors, Applicatives, and Monads: 10 Main Ideas](https://monadmadness.wordpress.com/2015/01/02/monoids-functors-applicatives-and-monads-10-main-ideas/)
+
+This article drops the dreaded "M" more than a few times ... but fear not. This post will help you to understand the key ideas behind building blocks of functional programming.
```
Comment (mtomas) [inline _posts/04-03-2015-digest21.md]: ...but have no fear about it.

File: _posts/04-03-2015-digest21.md [hunk 13]
```diff
@@ -0,0 +1,43 @@
...
+
+<p id="4"></p>
+* \[Mobile\] [How NativeScript Works](http://developer.telerik.com/featured/nativescript-works/) 
+
+Recently Facebook astonished us all, when they pushed Javascript onto mobile platforms while keeping it efficient. That's kind of a magic. Aren't you curious how does it really work? Take a look into the magical world of NativeScript.
+
+<p id="5"></p>
+* \[Android\] [An Investigation into Flow and Mortar](http://www.bignerdranch.com/blog/an-investigation-into-flow-and-mortar/) 
+
+Switching to Android from Scala might be painful. You have to drop many useful programming concepts, go back to writing bloated Java classes and the API feels just sad. Apparently the guys and gals at Square feel the same, so the published two nice libraries, that aim to simplify using Android fragments and structuring applications.
```
Comment (mtomas) [inline _posts/04-03-2015-digest21.md]: so they published

drop all commas please.

File: _posts/04-03-2015-digest21.md (hunk 12 above)
Comment (szabba) [inline _posts/04-03-2015-digest21.md]: What's wrong with the way it was? It was correct and succinct.
4 mar 2015 18:12 "Marek TOMAS" <notifications@github.com> napisał(a):

**Pull request merged**